src/Simulation/DocumentationGenerationStep/ProcessDocComments.cs

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Quantum.QsCompiler;
+using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.Documentation;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
+using Microsoft.Quantum.QsCompiler.SyntaxTree;
+using Microsoft.Quantum.QsCompiler.Transformations.Core;
+using Newtonsoft.Json.Linq;
+
+#nullable enable
+
+namespace Microsoft.Quantum.Documentation
+{
+    public class ProcessDocComments
+    : SyntaxTreeTransformation<ProcessDocComments.TransformationState>
+    {
+        public class TransformationState
+        { }
+
+        private string? OutputPath;
+
+        public ProcessDocComments(
+            string? outputPath = null
+        )
+        : base(new TransformationState())
+        {
+            OutputPath = outputPath;
+            // If the output path is not null, make sure the directory exists.
+            if (outputPath != null && !Directory.Exists(outputPath))
+            {
+                Directory.CreateDirectory(outputPath);
+            }
+
+            // We provide our own custom namespace transformation, and expression kind transformation.
+            this.Namespaces = new ProcessDocComments.NamespaceTransformation(this, outputPath);
+        }
+
+        public QsCompilation OnCompilation(QsCompilation compilation) =>
```

Comment:
Apply (inherited from the base class) would do that too.

Reply:
renamed to OnCompilation

Reply:
Does that imply that this method can be safely deleted?